layers/cmd/pipeline.go

```diff
@@ -0,0 +1,89 @@
+package cmd
+
+import (
+	"encoding/json"
+
+	"github.com/cloudprivacylabs/lsa/layers/cmd/cmdutil"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	"github.com/cloudprivacylabs/opencypher/graph"
+	"github.com/spf13/cobra"
+)
+
+type PipelineContext struct {
+	Context     *ls.Context
+	Graph       graph.Graph
+	Roots       []graph.Node
+	InputFiles  []string
+	currentStep int
+	steps       []Step
+}
+
+type Step interface {
+	Run(*PipelineContext) error
+	Next() error
+}
+
+var operations = make(map[string]func() Step)
+
+func (ctx *PipelineContext) Next() error {
+	ctx.currentStep++
+	if ctx.currentStep >= len(ctx.steps) {
+		return nil
+	}
+	err := ctx.steps[ctx.currentStep].Run(ctx)
+	ctx.currentStep--
+	return err
+}
+
+func init() {
+	rootCmd.AddCommand(pipelineCmd)
+	pipelineCmd.Flags().String("file", "", "Pipeline build file")
+	pipelineCmd.Flags().String("initialGraph", "", "Load this graph and ingest data onto it")
+}
+
+var pipelineCmd = &cobra.Command{
+	Use:   "pipeline",
+	Short: "run pipeline",
+	Args:  cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		file, err := cmd.Flags().GetString("file")
+		if err != nil {
+			failErr(err)
+		}
```

Comment:
Separate below this into a private func (file string, args []string) error
Test that func
Call that func from cobra cmd.